src/Umbraco.Core/Cache/DataTypeReadCache.cs

```diff
@@ -0,0 +1,64 @@
+using System.Collections.Concurrent;
+using Microsoft.Extensions.Caching.Memory;
+using Umbraco.Cms.Core.Events;
+using Umbraco.Cms.Core.Models;
+using Umbraco.Cms.Core.Notifications;
+using Umbraco.Cms.Core.Services;
+
+namespace Umbraco.Cms.Core.Cache;
+
+/// <summary>
+/// This cache is a temporary measure to reduce the amount of computational power required to deserialize and initialize <see cref="IDataType"/> when fetched from the main cache/database,
+/// because datatypes are fetched multiple times trough out a (backoffice content) request with a lot of content (or nested content) and each of these fetches initializes certain fields on the datatypes.
+/// </summary>
+/// <remarks>This cache should never be used in an CUD path as it can hold stale values.</remarks>
+public sealed class DataTypeReadCache : IDataTypeReadCache, INotificationHandler<DataTypeSavedNotification>,
+    INotificationHandler<DataTypeDeletedNotification>
```

Comment:
There's two very important things to know when implementing notification handlers for updating caches:
- Any `INotificationHandler` implementation won't get notified by Deploy (but `IDistributedCacheNotificationHandler` implementations do, see: https://github.com/umbraco/Umbraco-CMS/pull/14332);
- But even when using `IDistributedCacheNotificationHandler`, the notification will only be processed on the backoffice server (scheduling publisher) when using load balancing.

To properly refresh caches, use `INotificationHandler<DataTypeCacheRefresherNotification>` instead!

Reply:
@ronaldbarendse cacherefresher might happen not only on save operations, so it is not best thing to listen 🤔 as it might be cause by restart of umbraco and / or other operations which not create changes. but point of Deploy is valid. Not sure what will be approach here 🤔

Reply:
I do agree with @ronaldbarendse on this one. Chances are that sooner or later, someone will discover this cache (interface) and use it for some odd case on a subscriber instance. 

Alternatively make it internal - although I believe the implementations depending on the interface are public, which makes it a mute point.

Reply:
@kjac I think it should be internal as this service is temporary just to address performance issue and it should be investigated how to resolve issue fully and remove this specific service :)